File: 1996/4/3-4.py
#!/usr/bin/env python

import sys

msg = 1
while True:
    line = sys.stdin.readline()
    if line == "":
        break
    if line[-1] == "\n":
        line = line[:-1]
    rows,cols = map(int, line.split())
    ary = []
    for i in xrange(rows):
        ary.append(sys.stdin.readline().split())
    print "Message", msg, "=>",
    j = cols - 1
    for i in xrange(rows):
        print ary[i][j],
        j -= 1
        if j < 0:
            j = cols - 1
    print
    msg += 1

File: 1996/4/3-4.rb
#!/usr/bin/env ruby

file = File.open('buffy.dat')
msg = 1
while true
  begin
    line = file.readline
  rescue EOFError
    break
  end
  rows, cols = line.split.collect { |x| x.to_i }
  ary = []
  rows.times { ary << file.readline.split }
  print "Message #{msg} => "
  j = cols
  0...rows.each do |i|
    print ary[i][j-1]
    print ' '
    j -= 1
    j = cols if j == 0
  end
  puts
  msg += 1
end
